.github/workflows/cilium-integration-tests.yaml

```diff
@@ -0,0 +1,160 @@
+name: Cilium Integration Tests
+on:
+  push:
+    branches:
+      - master
+  pull_request: {}
+  issue_comment:
+    types:
+      - created
+env:
+  KIND_VERSION: v0.18.0
+  CILIUM_REPO_REF: master
+  CILIUM_CLI_TAG: latest
+
+jobs:
+  cilium-connectivity-tests:
+    timeout-minutes: 360
+    name: Cilium Connectivity Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Prepare variables for pushing to master
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+        run: |
+          echo "PROXY_IMAGE=quay.io/cilium/cilium-envoy" >> $GITHUB_ENV
+          echo "PROXY_TAG=${{ github.sha }}" >> $GITHUB_ENV
+
+      - name: Prepare variables for PR
+        if: github.event_name == 'pull_request'
```

Comment:
i realized that we need to filter out PR's (and comments on PR's) which aren't targeting `master`. as mentioned in another comment, we need to adapt once we're branching a new release branch from `master` and want to support it there too.

Reply:
adapted, that PR's based cilium integration tests are only triggered if the base branch of the PR is `master`. on other PR's there's still the possibility to use the variant with the issue comment on the PR.

in the future we might chose the right version of cilium & cilium-cli for other proxy release branches as well.